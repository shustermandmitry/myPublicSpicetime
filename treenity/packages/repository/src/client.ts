export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
