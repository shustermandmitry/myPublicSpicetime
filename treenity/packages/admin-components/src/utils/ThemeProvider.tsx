/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
