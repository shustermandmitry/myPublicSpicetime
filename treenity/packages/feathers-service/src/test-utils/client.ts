import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
