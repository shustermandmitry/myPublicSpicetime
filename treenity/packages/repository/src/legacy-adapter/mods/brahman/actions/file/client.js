import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
