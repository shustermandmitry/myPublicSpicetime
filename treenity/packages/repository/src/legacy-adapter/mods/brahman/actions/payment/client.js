import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
