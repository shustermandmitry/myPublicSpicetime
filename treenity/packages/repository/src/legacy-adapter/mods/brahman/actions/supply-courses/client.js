import './TelegramActionSupplyCourses.client';
import './TelegramCoursesOperator.client';
import './TelegramActionBecomeOperator.client';
