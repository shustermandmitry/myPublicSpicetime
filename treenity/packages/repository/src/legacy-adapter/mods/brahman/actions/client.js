import './addActionMenu';
import './back/TelegramActionBack.client';
import './broadcast/TelegramActionBroadcast.client';

import './call-command-by-param/TGActionCallCommandByParam.client';
import './crypto/TelegramActionGetRates.client';
// import './voice/video/SynhesizeAudioVideo';
import './emit-text/EmitText.client';
import './file/TelegramActionFileSend.client';
import './file/TelegramActionFileUpload.client';
import './file/TelegramFile.client';
import './forward/TelegramActionForward.client';
import './game/CurrentGame.client';
import './game/Game.client';
import './game/TelegramActionGame.client';
import './get-value/GetValue.client';
import './i18n/TGActionSelectLanguage.client';
import './if-else/TelegramActionIfElse.client';

import './keyword-select/TelegramActionKeywordSelect.client';
import './menu/TelegramActionMenu.client';
import './message/TelegramActionMessage.client';
import './on-error/TelegramActionOnError.client';
import './page/TelegramActionPage.client';
// import './pantaro/client';
import './params/TelegramActionParams.client';
import './payment/client';
import './question/TelegramActionQuestion.client';
import './removeMessage/TelegramActionRemove.client';
import './reset-history/ResetSession.client';
import './reset-session/ResetSession.client';
import './set-telegram-user/TGActionSetTelegramUser.client';
import './set-value/SetValue.client';

import './supply-courses/client';

import './tag/SetTag';
import './TelegramAction.client';
import './timers/TgActionSetTimer';

import './timers/timers.service.client';

import './translate/TString.client';
import './voice/tinkoff/TinkoffVoiceRecognition.client';
import './voice/tinkoff/TinkoffVoiceSynthesize.client';

import './voice/yandex/YandexVoiceRecognition.client';
import './voice/yandex/YandexVoiceSynthesize.client';

import './ym-statistics/TelegramPageStatistics.client';
