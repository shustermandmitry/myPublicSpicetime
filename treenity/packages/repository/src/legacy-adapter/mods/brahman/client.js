import './index.less';
import './bot/TelegramBot.client';
import './page/Page.client';
import './User.client';
import './Session.client';
import './actions/client';
import './translate/BotTranslate.client';
import './graph/PageGraph.client';
import './avatar/TegramAvatar.meta';
