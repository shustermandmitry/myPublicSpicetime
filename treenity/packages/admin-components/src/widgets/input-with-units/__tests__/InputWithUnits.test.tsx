/*
 * Copyright (c) 2024. Treenity Inc.
 */

import Icon from '@/components/icon';
import ThemeProvider from '@/utils/ThemeProvider';
import { fireEvent, render, waitFor, within } from '@testing-library/react';
import { createCookies, parseCookies, setCookie, CookieContext } from '@treenity/repository/client';
import { Button, Form } from 'antd';
import React from 'react';
import InputWithUnits from '../index';

const cookies = createCookies(parseCookies(document.cookie), setCookie);

describe('input-with-units component (Independent)', () => {
  const renderComponent = (props = {}) =>
    render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <InputWithUnits {...props} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

  it('renders input-with-units component', () => {
    const { container } = renderComponent({ value: '10px' });
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('displays the initial value and unit', () => {
    const { getByDisplayValue, getByText } = renderComponent({ value: '10px' });
    expect(getByDisplayValue('10')).toBeInTheDocument();
    expect(getByText('px')).toBeInTheDocument();
  });

  it('changes unit when a new unit is selected', async () => {
    const handleChange = jest.fn();
    const { container, getByRole } = renderComponent({
      value: '10px',
      onChange: handleChange,
    });

    const unitSpan = container.querySelector('.ant-dropdown-trigger') as HTMLElement;
    fireEvent.click(unitSpan);

    const menu = await waitFor(() => getByRole('menu'));
    const unitItem = within(menu).getByText('%').closest('li');
    fireEvent.click(unitItem!);

    expect(unitSpan).toHaveTextContent('%');
    expect(handleChange).toHaveBeenCalledWith('10%');
  });

  it('changes value when input value is changed', () => {
    const handleChange = jest.fn();
    const { getByDisplayValue } = renderComponent({ value: '20px', onChange: handleChange });
    const input = getByDisplayValue('20') as HTMLInputElement;

    fireEvent.change(input, { target: { value: '30' } });

    expect(input.value).toBe('30');
    expect(handleChange).toHaveBeenCalledWith('30px');
  });

  it('handles invalid input correctly', () => {
    const handleChange = jest.fn();
    const { getByDisplayValue } = renderComponent({ value: '20px', onChange: handleChange });
    const input = getByDisplayValue('20') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'abc' } });

    expect(input.value).toBe('');
    expect(handleChange).toHaveBeenCalledWith(undefined);
  });

  it('displays the slider when withRange prop is provided', () => {
    const { container } = renderComponent({ withRange: [0, 100], value: '50px' });
    expect(container.querySelector('.ant-slider')).toBeInTheDocument();
  });

  it('changes value when slider is used', () => {
    const handleChange = jest.fn();
    const { container } = renderComponent({
      withRange: [0, 100],
      value: '50px',
      onChange: handleChange,
    });

    const slider = container.querySelector('.ant-slider') as HTMLElement;
    fireEvent.mouseDown(slider, { clientX: 50 });
    fireEvent.mouseMove(slider, { clientX: 100 });
    fireEvent.mouseUp(slider);

    expect(handleChange).toHaveBeenCalledWith('100px');
  });

  it('shows icon when icon prop is provided', () => {
    const { container } = renderComponent({
      icon: <Icon name="add-folder_filled" />,
      value: '10px',
    });

    const icon = container.querySelector('.icon-add-folder_filled');
    expect(icon).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});

describe('input-with-units component (inside Form)', () => {
  const renderInsideForm = (props = {}, initialValues = {}, onSubmit = jest.fn()) =>
    render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <Form initialValues={initialValues} onFinish={onSubmit}>
            <Form.Item name="testInput">
              <InputWithUnits {...props} />
            </Form.Item>
            <Form.Item>
              <Button htmlType="submit">Submit</Button>
            </Form.Item>
          </Form>
        </ThemeProvider>
      </CookieContext.Provider>,
    );

  it('submits form with valid input', async () => {
    const handleSubmit = jest.fn();
    const { getByText, getByDisplayValue } = renderInsideForm(
      {},
      { testInput: '10px' },
      handleSubmit,
    );

    const input = getByDisplayValue('10') as HTMLInputElement;
    fireEvent.change(input, { target: { value: '35' } });
    fireEvent.click(getByText('Submit'));

    await waitFor(() => {
      expect(handleSubmit).toHaveBeenCalledWith({ testInput: '35px' });
    });
  });

  it('changes unit and value correctly within form', async () => {
    const { container, getByDisplayValue, getByRole } = renderInsideForm({}, { testInput: '10px' });

    const unitTrigger = container.querySelector('.ant-dropdown-trigger') as HTMLElement;
    expect(unitTrigger).toHaveTextContent('px');
    fireEvent.click(unitTrigger);

    const menu = await waitFor(() => getByRole('menu'));
    const unitItem = within(menu).getByText('%');
    fireEvent.click(unitItem);

    await waitFor(() => {
      expect(unitTrigger).toHaveTextContent('%');
    });

    const input = getByDisplayValue('10') as HTMLInputElement;
    fireEvent.change(input, { target: { value: '30' } });

    expect(input.value).toBe('30');
  });

  it('sets form value to undefined if input is empty', async () => {
    const handleSubmit = jest.fn();
    const { getByText, getByDisplayValue } = renderInsideForm(
      {},
      { testInput: '10px' },
      handleSubmit,
    );

    const input = getByDisplayValue('10') as HTMLInputElement;
    fireEvent.change(input, { target: { value: '' } });
    fireEvent.click(getByText('Submit'));

    await waitFor(() => {
      expect(handleSubmit).toHaveBeenCalledWith({ testInput: undefined });
    });
  });
});
