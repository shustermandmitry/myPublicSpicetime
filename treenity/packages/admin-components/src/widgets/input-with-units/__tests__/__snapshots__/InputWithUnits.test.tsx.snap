// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithUnits component (Independent) renders InputWithUnits component 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  width: 100%;
}

.emotion-1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  position: absolute;
  right: 0;
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  font-size: 8px;
  color: #A7A2BD;
  cursor: pointer;
  z-index: 2;
}

.emotion-2 span {
  padding: 0 4px;
}

.emotion-4 {
  height: 20px;
  font-size: 10px;
  border-radius: 4px;
  padding-inline: 6px;
  outline: none!important;
  font-weight: 500;
}

.emotion-4.emotion-4 .ant-input-prefix {
  -webkit-margin-end: 6px;
  margin-inline-end: 6px;
}

.emotion-4.emotion-4 i {
  font-size: 12px!important;
}

.emotion-4[type='number']::-webkit-inner-spin-button,
.emotion-4 input[type='number']::-webkit-inner-spin-button {
  display: none;
}

.emotion-4.ant-input {
  font-weight: 500!important;
}

.emotion-4 .ant-input-prefix i {
  color: #27ae60;
}

.emotion-4 .ant-input-prefix {
  -webkit-margin-end: 8px;
  margin-inline-end: 8px;
}

.emotion-4 i {
  color: #d9d9d9;
  -webkit-transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-4.ant-input-affix-wrapper-focused i {
  color: #27ae60;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) {
  background-color: #eb57570a;
  border-width: 1px;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) i {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i:before,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) i:before {
  content: '\\e91c';
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) input {
  background: transparent;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-moz-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-status-error:not(.ant-input-disabled) input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-status-error:not(.ant-input-disabled) input::-moz-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-status-error:not(.ant-input-disabled) input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) input,
.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::placeholder,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) input::placeholder {
  color: #eb5757;
}

.emotion-4 .ant-input-group-addon {
  border-left: 0;
}

.emotion-4 i {
  color: #d8d8da;
  font-size: 12px!important;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <span
          class="ant-dropdown-trigger"
        >
          px
        </span>
      </div>
      <input
        aria-label="Value input"
        class="ant-input ant-input-sm emotion-3 ant-input-outlined input-with-units emotion-4"
        placeholder="0"
        type="text"
        value="10"
      />
    </div>
  </div>
</div>
`;

exports[`InputWithUnits component (Independent) shows icon when icon prop is provided 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4px;
  width: 100%;
}

.emotion-1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  position: absolute;
  right: 0;
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  font-size: 8px;
  color: #A7A2BD;
  cursor: pointer;
  z-index: 2;
}

.emotion-2 span {
  padding: 0 4px;
}

.emotion-4 {
  height: 20px;
  font-size: 10px;
  border-radius: 4px;
  padding-inline: 6px;
  outline: none!important;
  font-weight: 500;
}

.emotion-4.emotion-4 .ant-input-prefix {
  -webkit-margin-end: 6px;
  margin-inline-end: 6px;
}

.emotion-4.emotion-4 i {
  font-size: 12px!important;
}

.emotion-4[type='number']::-webkit-inner-spin-button,
.emotion-4 input[type='number']::-webkit-inner-spin-button {
  display: none;
}

.emotion-4.ant-input {
  font-weight: 500!important;
}

.emotion-4 .ant-input-prefix i {
  color: #27ae60;
}

.emotion-4 .ant-input-prefix {
  -webkit-margin-end: 8px;
  margin-inline-end: 8px;
}

.emotion-4 i {
  color: #d9d9d9;
  -webkit-transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: color 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

.emotion-4.ant-input-affix-wrapper-focused i {
  color: #27ae60;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) {
  background-color: #eb57570a;
  border-width: 1px;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) i {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper i:before,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) i:before {
  content: '\\e91c';
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) input {
  background: transparent;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::-moz-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-status-error:not(.ant-input-disabled) input::-webkit-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-status-error:not(.ant-input-disabled) input::-moz-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-status-error:not(.ant-input-disabled) input:-ms-input-placeholder {
  color: #eb5757;
}

.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) input,
.emotion-4.ant-input-affix-wrapper-status-error:not(.ant-input-affix-wrapper-disabled):not(
      .ant-input-affix-wrapper-borderless
    ).ant-input-affix-wrapper input::placeholder,
.emotion-4.ant-input-status-error:not(.ant-input-disabled) input::placeholder {
  color: #eb5757;
}

.emotion-4 .ant-input-group-addon {
  border-left: 0;
}

.emotion-4 i {
  color: #d8d8da;
  font-size: 12px!important;
}

.emotion-5 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <span
          class="ant-dropdown-trigger"
        >
          px
        </span>
      </div>
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-sm emotion-3 ant-input-outlined input-with-units emotion-4"
      >
        <span
          class="ant-input-prefix"
        >
          <i
            class="icon-add-folder_filled emotion-5"
          />
        </span>
        <input
          aria-label="Value input"
          class="ant-input ant-input-sm emotion-3"
          placeholder="0"
          type="text"
          value="10"
        />
      </span>
    </div>
  </div>
</div>
`;
